src/FSLibrary/MaxTPSTest.fs

```diff
@@ -0,0 +1,231 @@
+// Copyright 2024 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+module MaxTPSTest
```

Comment:
I recall us chatting about some initial numbers we got from this test, and they seem a bit too low. What kind of failure are we observing in this test that causes it to fail? Is it tx queue getting overloaded, nodes going out of sync, or something else?

Reply:
For context, our classic test pushed around 900 TPS (this is without simulated apply time though), so at least for classic I'd expect to see comparable numbers.

Reply:
Max TPS is around 90 TPS in the mixed test. It's hard to tell what exactly the issue is. One common failure mode is transactions generating with bad sequence numbers causing loadgen to fail. This is a failure seen in a run with TPS set to 110:
```
2024-06-07T00:25:48.206 GATFR [LoadGen INFO] tx rejected 'ERROR': ===> soroban, {
    "TransactionResult": {                                                      
        "feeCharged": 839775,                                                   
        "result": {                                                             
            "code": "txBAD_SEQ"                                                 
        },                                                                      
        "ext": {                                                                
            "v": 0                                                              
        }                                                                       
    }                                                                           
}                                                                               
```
[After enough of these in a row, loadgen fails early](https://github.com/stellar/stellar-core/blob/b3aeb14cc798f6d11deb2be913041be916f3b0cc/src/simulation/LoadGenerator.cpp#L840). I'm not sure why the sequence numbers are getting out-of-sync.

Something weird I've noticed is that as the TPS gets lower and closer to passing, this specific error goes away and supercluster marks loadgen as failed after all nodes finish generating all of their load, hinting at some kind of synchronization or invariant error.
 
I suspect the underlying issue is the same and at higher TPS values it just overwhelms something to the point that loadgen fails early.

Reply:
One thing I'd like to try is to play around with the ratios. We should expect to see TPS go up with fewer soroban txs in blended mode. I also wonder whether the ratio of upload to invoke makes a different. I'd assume that upload puts a lot more strain on the overlay.

Reply:
The metrics look mostly normal, but it looks like `ledger.age.closed` had some high values, peaking around 26s:
```
,{                                                                              
"boundary": 1.79769e+308,                                                       
"type":"timer",                                                                 
"count":3,                                                                      
"event_type":"calls",                                                           
"rate_unit":"s",                                                                
"mean_rate":0.00234157,                                                         
"1_min_rate":3.21809e-06,                                                       
"5_min_rate":0.00145292,                                                        
"15_min_rate":0.0019286,                                                        
"duration_unit":"ms",                                                           
"min":25074.1,                                                                  
"max":26276.8,                                                                  
"mean":25529.3,                                                                 
"stddev":652.404,                                                               
"sum":76588,                                                                    
"median":0,                                                                     
"75%":0,                                                                        
"95%":0,                                                                        
"98%":0,                                                                        
"99%":0,                                                                        
"99.9%":0,                                                                      
"100%":0                                                                        
}     
```

That being said, looking into some other nodes that failed early in the run I see a few transactions failing to be added to the queue with `txINSUFFICIENT_FEE`:
```
2024-06-07T00:30:44.245 GDOKV [LoadGen INFO] tx rejected 'ERROR': ===> soroban, {
    "TransactionResult": {                                                      
        "feeCharged": 2819102,                                                  
        "result": {                                                             
            "code": "txINSUFFICIENT_FEE"                                        
        },                                                                      
        "ext": {                                                                
            "v": 0                                                              
        }                                                                       
    }                                                                           
} 
```

I need to add some additional debug info to stellar-core to figure out why these are flagged that way (could be poor estimates in loadgen, or surge pricing kicking in, or something else).

Reply:
Could you please try running the test on our normal cluster so we can use the grafana dashboard to look at the metrics on all nodes across the whole run. It's a bit hard to evaluate scraped metrics for every node; it could be just one node causing issues. Also each individual scrape only gives us info for the last 30 seconds, so we might be missing information.

Reply:
Well, this is weird. I'm seeing much better performance on the old cluster. It just passed a 200 TPS run. I'm going to let it run to see how far it gets. Maybe I've got something misconfigured on the new cluster...

Reply:
I'm also using a slightly different set of flags (to better match the ones in jenkins), so that could also be it. I definitely have a few more things to investigate before I can narrow it down to the new cluster.

Reply:
Huh weird! In case this helps, I just checked our results for the classic max TPS test, and looks like it didn't change much after switching to the new cluster, so this might be related to mission configuration.

Reply:
Good to know!